src/main/java/com/avispl/symphony/dal/infrastructure/gateway/haivision/srt/HaivisionGatewayCommunicator.java

```diff
@@ -0,0 +1,683 @@
+	 */
+	private void initialCookieSession() throws FailedLoginException {
+		try {
+			Map<String, String> bodyRequest = new HashMap<>();
+			bodyRequest.put("username", this.getLogin());
+			bodyRequest.put("password", this.getPassword());
+			JsonNode response = this.doPost(HaivisionCommand.API_SESSION, bodyRequest, JsonNode.class);
+			if (response != null && response.has(HaivisionConstant.RESPONSE) && response.get(HaivisionConstant.RESPONSE).has(HaivisionConstant.SESSION_ID)) {
+				this.authenticationCookie = response.get(HaivisionConstant.RESPONSE).get(HaivisionConstant.SESSION_ID).asText();
+				return;
+			}
+			this.authenticationCookie = HaivisionConstant.EMPTY;
+			throw new ResourceNotReachableException("Unable to retrieve the authorization token, endpoint not reachable.");
+		} catch (FailedLoginException ex) {
+			throw new FailedLoginException("Unable to login. Please check device credentials");
+		} catch (Exception e) {
+			throw new ResourceNotReachableException("Unable to retrieve the authorization token, endpoint not reachable");
+		}
+	}
+
+	/**
+	 * Verifies if the current authentication cookie session is still valid.
+	 *
+	 * This method sends a request to the server to validate the current session.
+	 * If the session is invalid or an error occurs, a new session is initialized.
+	 *
+	 * @throws Exception if an error occurs during the validation of the cookie session.
+	 */
+	private void checkValidCookieSession() throws Exception {
+		try {
+			JsonNode response = this.doGet(HaivisionCommand.API_SESSION, JsonNode.class);
+			if (response == null || response.has(HaivisionConstant.ERROR)) {
+				initialCookieSession();
+			}
+		} catch (Exception e) {
+			logger.info("Invalid session ID " + this.authenticationCookie);
+			initialCookieSession();
+		}
+	}
+
+	/**
+	 * Deletes the current authentication cookie session from the server.
+	 *
+	 * This method sends a request to the server to delete the current session ID.
+	 * After deleting the session, the local authentication cookie is reset to an empty value.
+	 */
+	private void deleteCookieSession() {
+		try {
+			this.doDelete(HaivisionCommand.API_SESSION);
+			logger.info("Delete session ID " + this.authenticationCookie);
+		} catch (Exception e) {
+			logger.info("Error white delete session ID " + this.authenticationCookie);
+		} finally {
+			this.authenticationCookie = HaivisionConstant.EMPTY;
+		}
+	}
+
+	/**
+	 * Retrieves monitoring properties for a device and populates the cache with the device's information.
+	 *
+	 * @throws ResourceNotReachableException if an error occurs when retrieving the device information.
+	 */
+	private void retrieveMonitoringProperties() {
+		try {
+			JsonNode response = this.doGet(HaivisionCommand.GET_DEVICE_INFO, JsonNode.class);
+			if (response != null && response.isArray()) {
+				JsonNode deviceInfo = response.get(0);
+				deviceId = deviceInfo.get("_id").asText();
```

Comment:
get() may produce npe here, are we sure it's there? 
at("/...") can be used instead, so NPE won't be produced if there's no such node present